source/opt/folding_rules.cpp

```diff
@@ -2079,21 +2080,26 @@ uint32_t GetNumberOfElements(const analysis::Type* type) {
     return static_cast<uint32_t>(struct_type->element_types().size());
   }
   if (auto* array_type = type->AsArray()) {
-    return array_type->length_info().words[0];
+    if (array_type->length_info().words[0] == 0 &&
```

Comment:
Can you use the `Array::LengthInfo::kConstant` enum for readability?

Reply:
Good call.